### PR TITLE
Refacto main / Added support for plantuml and mermaid file

### PR DESCRIPTION
**Pull request by leoTostain**
Re-factorized the code with a method to get the list of packages and added a fix in the pom.xml to remove a warning (again) when making the uber/jar.

### CODE REVIEW

**Comment by tvelu77** [comment]
added support for .plantuml and .mermaid files
fixed a bug where the program threw a NPE when we are using mmd or pu

File: src/main/java/com/github/forax/umldoc/Main.java [hunk 1]
```diff
@@ -84,11 +74,29 @@ public static void main(String[] args) {
     }
   }
 
+  private static List<Package> getPackage(ModuleFinder finder) {
+    var module = finder.findAll().stream().findFirst();
+
+    if (module.isEmpty()) {
+      System.err.println("Couldn't find a Module");
+      System.exit(-1);
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/Main.java:82]: System.exit() should only be used in the main()

File: src/main/java/com/github/forax/umldoc/Main.java [hunk 2]
```diff
@@ -84,11 +74,29 @@ public static void main(String[] args) {
...
+      return null;
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/Main.java:83]: If there is an error, throw an exception

File: src/main/java/com/github/forax/umldoc/Main.java [hunk 3]
```diff
@@ -74,21 +78,17 @@ public static void main(String[] args) {
...
+      throw new IllegalStateException("Couldn't find a Module");
     }
 
     try {
       return ModuleScrapper.scrapModule(module.get());
     } catch (IOException e) {
-      System.err.println("Couldn't get the list of Packages : " + e.getMessage());
-      System.exit(-1);
-      return null;
+      throw new IllegalStateException("Couldn't get the list of Packages : " + e.getMessage());
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/Main.java:91]: here, you should chain the exception instead of using getMessages().

**Pull request merged**